core/cafe_api.go

```diff
@@ -280,23 +280,40 @@ func (c *cafeApi) service(g *gin.Context) {
 		g.String(http.StatusBadRequest, err.Error())
 		return
 	}
+	if rpmes != nil {
+		res, err := proto.Marshal(rpmes)
+		if err != nil {
+			g.String(http.StatusBadRequest, err.Error())
+			return
+		}
 
-	if rpmes == nil {
-		g.Status(200)
+		g.Render(http.StatusOK, render.Data{Data: res})
 		return
 	}
 
-	// send out response msg
-	log.Debugf("responding with %s to %s", rpmes.Message.Type.String(), mPeer.Pretty())
-
-	res, err := proto.Marshal(rpmes)
-	if err != nil {
-		g.String(http.StatusBadRequest, err.Error())
-		return
-	}
+	// handle the message as a JSON stream
+	rpmesCh, errCh, cancel := c.node.cafe.HandleStream(mPeer, pmes)
+	g.Stream(func(w io.Writer) bool {
```

Comment:
Stream search results back down to the API caller...

Reply:
love it